src/entities/event/api/eventApi.ts

```diff
@@ -10,13 +10,19 @@ interface ApiResponse<T> {
 
 export const eventApi = baseApi.injectEndpoints({
   endpoints: (build) => ({
-    getEvents: build.query<ApiResponse<IEvent[]>, void>({
-      query: () => ({
+    getEvents: build.query<IApiResponse<IEvent[]>, string | null>({
+      query: (filter) => (filter ? {
         url: '/events/',
-        method: 'GET'
+        method: 'GET',
+        params: {
+          search: filter
+        }
+      } : {
```

Comment:
For now we are only using the serach filter, but this endpoint can be used with multiple filters.

```suggestion
// Add EventFilter type
EventFilter {
search?: string
}
...
// Use it as endpoint arg
....
query: (filter: EventFilter) => ({
        url: '/events/',
        method: 'GET',
        params: {
          search: filter.search
        }
    ....
```

We can use empty default values for filter parameters and the backend will ignore them, am I right @jakkemomo ?

Reply:
> We can use empty default values for filter parameters and the backend will ignore them, am I right @jakkemomo ?

I checked in the backend, this is correct, at least for the search argument.